### ISSUE TITLE
Syntax error with parentheses in "if" statement followed by function call

### ISSUE THREAD

**Issue by rpsirois** [jsy-lang/jsy-lang-docs#2]
Using an inline conditional `(a || b)` in the signature for a conditional seems to trick the syntax parser into leaving parentheses open when using at `@` syntax afterward.

Reproduce ([playground](https://jsy-lang.github.io/#JYMwBAFAtgngygFwE7AHYHMwB8tgOR4CUAdAM4ICGSCpA6sAgBZgAC+FeYAXFwFBhgAxgHtUpYQBsApsQnDMbPMDAAjJMIDWUsAzxA)) (using parentheses with function call):
```js
if (myString || '').startsWith @ 'a' ::
  console.log @ 'i broke it'
```

Expected behavior: Close function arguments after `@` then begin conditional block after `::`.

**Comment by rpsirois** [comment]
For reference, a quick fix is: 
```js
if @ (a||b).fn @ arg ::
  ...
```

**Comment by shanewholloway** [comment]
JSY's keyword control parser uses the presence of an opening left parenthesis to trigger JavaScript compatible parsing, allowing JSY to be a superset of JavaScript. When no opening parenthesis is encountered, JSY parser expects a `::` transition to move to keyword body.

The quick fix is the recommended solution, positively signaling to the parser that it should look for `::` to transition. (It was actually the primary syntax for keywords with conditionals for a while.) Other options include the "truth" double-not operator  `!! (a || b)`

In other words, this is a mismatch of user expectations to the superset of JavaScript feature. How might we better communicate this behavior to align user expectations?

**Comment by rpsirois** [comment]
Ah thank you for the explanation. That makes sense of course but certainly subtle! I believe the Syntax Error as it stands is the proper error. I'll look for some more explainers in the docs, thanks!
